pkg/engine/planner/logical/builder_test.go

```diff
@@ -0,0 +1,65 @@
+package logical
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/loki/v3/pkg/engine/internal/types"
+)
+
+func TestBuilderParse(t *testing.T) {
+	t.Run("Builder adds Parse after MakeTable", func(t *testing.T) {
```

Comment:
I don't think this test is actually validating any business logic. The assertion is "Builder adds Parse after MakeTable," but that's based on the test calling the builder in that order itself. 

We can remove this in favour of testing the planner API.

Reply:
agreed, it's just testing the test setup itself, will move planner tests higher through the the planning API